txnbuild/transaction.go

```
@@ -346,6 +346,65 @@ func (t *Transaction) Base64() (string, error) {
 	return marshallBase64(t.envelope, t.signatures)
 }
 
+// BalanceIDs returns a list of balance IDs for all possible CreateClaimableBalance
+// operations within this transaction.
+func (t *Transaction) BalanceIDs() []string {
+	var balances []string
+	for opIndex, operation := range t.operations {
+		creation, ok := operation.(*CreateClaimableBalance)
+		if !ok {
+			continue
+		}
+
+		// Use the operation's source account or the transaction's source if not.
+		var account Account
+		if creation.SourceAccount == nil {
+			account = &t.sourceAccount
+		} else {
+			account = creation.GetSourceAccount()
+		}
+
+		seq, err := account.GetSequenceNumber()
+		if err != nil {
+			panic(errors.Wrap(err, "failed to query account sequence num"))
```

Can we return these errors instead of gobbling them?

---

Good call, fixed! This was originally a product of the fact that I didn't want a half-processed array returned, but with the refactor to be specific to one operation, returning errors fits perfectly into the design.